lib/money/money.rb

```diff
@@ -209,17 +206,24 @@ def split(num)
   end
 
   private
-  # poached from Rails
-  def value_to_decimal(value)
-    # Using .class is faster than .is_a? and
-    # subclasses of BigDecimal will be handled
-    # in the else clause
-    if value.class == BigDecimal
-      value
-    elsif value.respond_to?(:to_d)
-      value.to_d
+
+  def value_to_decimal(num)
+    if num.respond_to?(:to_d)
+      num.is_a?(Rational) ? num.to_d(16) : num.to_d
```

Comment:
`Rational#to_d` actually requires a precision to be passed and doesn't provide a default.